Taken from upstream commit 63c1d0ceb35b9dc41da1f7114e2b1adc204ec097
"build: migrate to Gradle 9.6.1"

--- build-logic/basics/src/main/kotlin/testng.reproducible-builds.gradle.kts.orig	2026-01-21 19:03:44
+++ build-logic/basics/src/main/kotlin/testng.reproducible-builds.gradle.kts	2026-10-30 23:37:36
@@ -2,6 +2,6 @@
     // Ensure builds are reproducible
     isPreserveFileTimestamps = false
     isReproducibleFileOrder = true
-    dirMode = "775".toInt(8)
-    fileMode = "664".toInt(8)
+    dirPermissions { unix("775") }
+    filePermissions { unix("664") }
 }
--- build-logic/build-parameters/build.gradle.kts.orig	2026-01-21 19:03:44
+++ build-logic/build-parameters/build.gradle.kts	2026-10-30 23:39:23
@@ -1,6 +1,6 @@
 plugins {
     id("org.gradlex.build-parameters") version "1.4.4"
-    id("com.github.vlsi.gradle-extensions") version "1.90"
+    id("com.github.vlsi.gradle-extensions") version "3.0.2"
     id("build-logic.kotlin-dsl-gradle-plugin")
 }
 
--- build-logic/jvm/build.gradle.kts.orig	2026-01-21 19:03:44
+++ build-logic/jvm/build.gradle.kts	2026-10-30 23:40:17
@@ -6,6 +6,6 @@
     api(projects.buildParameters)
     api(projects.basics)
     api(projects.codeQuality)
-    api("com.github.vlsi.gradle-extensions:com.github.vlsi.gradle-extensions.gradle.plugin:1.90")
+    api("com.github.vlsi.gradle-extensions:com.github.vlsi.gradle-extensions.gradle.plugin:3.0.2")
     api("org.jetbrains.kotlin.jvm:org.jetbrains.kotlin.jvm.gradle.plugin:2.3.0")
 }
--- build-logic/publishing/build.gradle.kts.orig	2026-01-21 19:03:44
+++ build-logic/publishing/build.gradle.kts	2026-10-30 23:42:02
@@ -4,8 +4,8 @@
 
 dependencies {
     api(projects.jvm)
-    api("com.github.vlsi.gradle-extensions:com.github.vlsi.gradle-extensions.gradle.plugin:1.90")
+    api("com.github.vlsi.gradle-extensions:com.github.vlsi.gradle-extensions.gradle.plugin:3.0.2")
     api("com.gradleup.nmcp:com.gradleup.nmcp.gradle.plugin:1.4.1")
-    api("com.github.johnrengelman.shadow:com.github.johnrengelman.shadow.gradle.plugin:8.1.1")
+    api("com.gradleup.shadow:com.gradleup.shadow.gradle.plugin:9.6.1")
     api("org.jetbrains.kotlin:kotlin-gradle-plugin")
 }
--- build-logic/publishing/src/main/kotlin/buildlogic/OptionalFeaturesExtension.kt.orig	2026-01-21 19:03:44
+++ build-logic/publishing/src/main/kotlin/buildlogic/OptionalFeaturesExtension.kt	2026-10-30 23:44:30
@@ -41,7 +41,8 @@
     init {
         // By default, shade all modules from the current build
         shadedDependenciesFilter.convention {
-            it.owner.let { id -> id is ProjectComponentIdentifier && id.build.isCurrentBuild }
+            // BuildIdentifier.isCurrentBuild() was removed in Gradle 9; the root build's path is ":".
+            it.owner.let { id -> id is ProjectComponentIdentifier && id.build.buildPath == ":" }
         }
     }
 
--- build-logic/settings.gradle.kts.orig	2026-01-21 19:03:44
+++ build-logic/settings.gradle.kts	2026-10-30 23:48:28
@@ -1,3 +1,9 @@
+pluginManagement {
+    // Gradle 9 resolves plugins from included builds only when they are included
+    // via pluginManagement. build-logic-commons provides build-logic.kotlin-dsl-gradle-plugin.
+    includeBuild("../build-logic-commons")
+}
+
 dependencyResolutionManagement {
     repositories {
         gradlePluginPortal()
@@ -8,7 +14,6 @@
 
 enableFeaturePreview("TYPESAFE_PROJECT_ACCESSORS")
 
-includeBuild("../build-logic-commons")
 include(":basics")
 include(":build-parameters")
 include(":code-quality")
--- testng/testng-build.gradle.kts.orig	2026-01-21 19:03:44
+++ testng/testng-build.gradle.kts	2026-10-30 23:51:41
@@ -11,7 +11,7 @@
     // third-party dependencies would be left as regular dependencies
     optionalFeatures {
         shadedDependenciesFilter.set {
-            it.owner.let { id -> id is ProjectComponentIdentifier && id.build.isCurrentBuild }
+            it.owner.let { id -> id is ProjectComponentIdentifier && id.build.buildPath == ":" }
         }
 
         create("guice") {
@@ -35,8 +35,8 @@
     manifest {
         // providers.gradleProperty does not work
         // see https://github.com/gradle/gradle/issues/14972
-        val name = rootProject.findProperty("project.name")
-        val vendor = rootProject.findProperty("project.vendor.name")
+        val name = rootProject.property("project.name").toString()
+        val vendor = rootProject.property("project.vendor.name").toString()
         attributes(
             // Java 9 module name
             "Automatic-Module-Name" to project.group,
@@ -48,7 +48,7 @@
             "Bundle-Vendor" to vendor,
             // See http://docs.osgi.org/specification/osgi.core/7.0.0/framework.module.html#i2654895
             "Bundle-License" to "Apache-2.0",
-            "Bundle-Description" to project.description,
+            "Bundle-Description" to project.description.orEmpty(),
             "Bundle-Version" to project.version.toString().removeSuffix("-SNAPSHOT"),
             //TestNG loads classes "by name" from configuration files, this allows to load such classes without need to know the exact package name
             "DynamicImport-Package" to "*",
